Adding release/pre-release detection

wallentx opened the pull request:
Swap the tag creation trigger for `release.published` trigger, so that we can leverage conditionals based on `release` vs `pre-release`

wallentx committed:
Adding release detection

wallentx committed:
Fine tuning release scenario, and including conditional for bladebit

cmmarslender commented on .github/workflows/build-linux-arm64-installer.yml:
```diff
@@ -101,12 +102,22 @@ jobs:
         chmod +x "$GITHUB_WORKSPACE/bladebit/bladebit"
 
+    - name: Get latest prerelease bladebit plotter
+      if: github.event.release.published && '!github.event.release.prerelease'
```
```suggestion
      if: github.event.release.published && 'github.event.release.prerelease'
```

I think for prereleases we'd want the prerelease bladebit (I think this logic might be backwards?)

cmmarslender commented on .github/workflows/upload-pypi-source.yml:
```diff
@@ -6,8 +6,8 @@ on:
       - 'release/**'
-    tags:
-        - '**'
+  release:
```
For the last step in here (when we publish to pypi) I'm wondering if theres a way we can mark these as prerelease there when its a pre-release

wallentx replied:
https://github.com/pypa/gh-action-pypi-publish
Just looking at this, it doesn't say anything specific, but if it handles tags, we could just maybe enforce a tag standard syntax for pre-release?

wallentx replied:
Actually, this may answer it:
https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#pre-release-versioning
What I gather from that is that it is considered a pre-release based on the contents of the tag.
https://pypi.org/project/chia-blockchain/#history
It looks like it even already has some pre-releases flagged. I think this should be good to go, so long as our tagging format for pre-releases follows one of the conventions that pypi sets.

wallentx committed:
Fixing bladebit conditional check

wallentx committed:
Ensuring Get tag name job only runs for tagged events

wallentx committed:
Removing double if on linux arm workflow

wallentx committed:
Removing double if on macos workflow

wallentx committed:
Adding newly tested changes for release/prerelease

wallentx committed:
Fixing deb installer

wallentx committed:
Testing windows installer change

wallentx committed:
Update build-windows-installer.yml

wallentx committed:
Update build-windows-installer.yml

wallentx committed:
Update build-windows-installer.yml

wallentx committed:
Update build-windows-installer.yml

wallentx committed:
Update build-windows-installer.yml

wallentx committed:
Update build-windows-installer.yml

wallentx committed:
Update build-windows-installer.yml

wallentx committed:
Update build-windows-installer.yml

wallentx committed:
Update build-windows-installer.yml

wallentx committed:
testing without setting windows version

wallentx committed:
Wallentx/pr workflow2 (#5)

* Fix duplicate subscriptions.

* fixup

* Bump chia-blockchain-gui to 55e776c875f27d2e80fed0105691ff8183040e18 (#13090)

Fixes an exception in local storage when non-existent items are accessed.

* some rpc reference data output code (commented out)

* renames

* Address linting.

* Update test to reflect the recent change in ReorgProtocol.

* include wallet maker offer rpc call in transaction rollback region

* Update data_layer.py

* Adding changelog (#13121)

* Adding changelog

* Fixing changelog indentation

* roll back pending roots on cancellation (but broken)

* (fixed up)

* bump data layer test timeout to 55

* util: Introduce `get_key`/`get_keys` in `Keychain` + `KeychainServer` (#12830)

* Generalize NFT offers (#12945)

* Generalize NFT offers

* Write tests for complex NFT1 offers

* Combine two loops

* isort

* Save one block farmed

* Increase timeouts

* black

* merge related fix

* avoid locking up the main thread by using an asyncio.Future to wait for the worker processes in weight proof validation (#13118)

* now that all uses of DBWrapper have been updated to DBWrapper2, remove the old one (#13116)

* Adding changelog (#13121) (#13135)

* Adding changelog

* Fixing changelog indentation

* Fix several bugs with untrusted sync, and correct sync status (#13133)

* Hopefully fix bug with untrusted sync

* Try removing the cache

* Add cache back, and don't rollback

* Compare to finished_sync_up_to

* Fix plotnfts, and tests

* Async function

* fix wallet blockchain tests

* Fix test, and improve detection of synced

* Correct sync status

* Fix more tests

* Fix more tests

* Fix NFT tests

* We did not await the async function (#13154)

* dependency-review (#13153)

* tweak output from task-profiler (#13137)

* Replace `electron-packager` with `electron-builder` (#12953)

* Install electron-builder

* Replaced electron-packager with electron-builder for windows

* Upgraded electron-builder version for mac

* Replaced electron-packager with electron-builder for Linux and integreated deb/rpm dir

* Updated mac installer job

* Fixed app.asar location

* Fixed dmg name for arm64 mac

* Updated installer-version.py

* Updated Linux(deb) installer CI job

* Updated Linux(rpm) installer CI job

* Fixed Linux(rpm) install CI issue

* Revert to use `electron-packager` instead of `electron-builder` for arm64 .deb installer

* Fixed lint errors

* Fixed rpm installer job issue

* Updated Windows installer CI job

* Fixed rpm installer CI job

* Simplified arm64 build

* Added -y option to `apt install` command

* Added `dmg-license` as npm build dependency for MacOS

* Attempt to set product name appropriately for each platform

* Specified productNames in commands

* Fixed arm64 installer issue

* Stopped to create link '/opt/chia/daemon'

* Updated desktop app name for linux installers

* Updated packageName for linux installers

* Abort build jobs when 'cd' fails

* Updated the GUI submodule ref (main:5e52b7f53478d4f39b4a0f0203f41e31dd53aee9)

Co-authored-by: Jeff Cruikshank <jeff@chia.net>

* update _peak_height after committing the block to the DB (#13157)

* When adding coins to the wallet, lookup the coin records in bulk, rather than one at a time (#13142)

* Revert "now that all uses of DBWrapper have been updated to DBWrapper2, remove the old one" (#13145)

This reverts commit 2d9672c4e75290b1ca731c7225af01b153a4672e.

* Fix comments in initial config where puzzle hash should be receive address (#13146)

* Add atari CLI help.

* Add atari CLI.

* Update chia/cmds/data.py

Co-authored-by: Kyle Altendorf <sda@fstab.net>

* Update chia/cmds/data.py

Co-authored-by: Kyle Altendorf <sda@fstab.net>

* Update chia/cmds/data.py

Co-authored-by: Kyle Altendorf <sda@fstab.net>

* Update chia/cmds/data.py

Co-authored-by: Kyle Altendorf <sda@fstab.net>

* Update chia/cmds/data.py

Co-authored-by: Kyle Altendorf <sda@fstab.net>

* Update chia/cmds/data.py

Co-authored-by: Kyle Altendorf <sda@fstab.net>

* Lint.

* atari - drop dead fingerprint options

* Reflect the change on datalayer related code.

* Fixup.

* Don't try to sign mac installers when secrets are unavailable (#13166)

* util: Implement key label support for `FileKeyring` and `KeyringWrapper` (#12843)

* util: Implement key label support for `FileKeyring` and `KeyringWrapper`

* Restrict trailing/leading whitespaces in the label

* Bump `MAX_LABEL_LENGTH` to 65

* Additional label tests

* Drop duplicated `KeychainFingerprintNotFound` definition

* add royalty calculation method and RPC (#13129)

* Generalize NFT offers

* Write tests for complex NFT1 offers

* Combine two loops

* isort

* Save one block farmed

* Increase timeouts

* black

* merge related fix

* add royalty calculation method and RPC

* flake8

* Fix the CLI for royalties

* royalty_pts -> royalty_percentage

* minor test fix

* remove all_puzzle_hashes from wallet puzzle store (#13159)

* simplify WalletStateManager.coin_added(). Make sure all callers ensure the coin isn't in the DB already, and always pass in the coin name (#13144)

* update gui for atari changes

* enable mypy for a some wallet files (#13185)

* enable mypy for chia/wallet/trading/offer.py

* enable mypy for chia/wallet/trade_manager.py

* track last_derivation_index in memory for each wallet (and across wallets) rather than performing DB lookups unconditionally (#13155)

* introduce a Protocol for puzzle drivers, for increased type checking (#13183)

* update gui for release/1.6.0 take 2

* core.data_layer parallel = True (#13197)

* don't run the get_offered_coins post-init check. It's very expensive (#13205)

* skip parsing (and computing) offers just to get to the involved coins, instead use the DB field for coins_of_interest (#13213)

* implement a more efficient Program.uncurry() (#13204)

* Debug get tree hash (#13218)

* log clvm programs with format string instead of f-string

* avoid redundant calls to get_tree_hash() in the NFT wallet

* only compute the tree hashes for debug logging when actually logging

* Fill missing services in config for data layer (#13220)

* fill missing services in config for data layer

* adjust

* here too

* util: Implement key label support for `Keychain` and `KeychainServer` (#12883)

* util: Implement key name support for `Keychain` and `KeychainServer`

* Re-raise if adding the key fails

* Call `G1Element.get_fingerprint` only once

* use rust parser for Program and SerializedProgram.to_program() (#13203)

* benchmark for parsing Offer (#13212)

* Puzzle driver type annotations (#13190)

* annotate CAT outer puzzle driver

* add type annotation to transfer_program_puzzle

* add type annotations to singleton_outer_puzzle

* add type annotations to ownership_outer_puzzle

* add type annotations to metadata_outer_puzzle

* remove unused _asset_id field in puzzle drivers

* Fix missing wallet `state_changed` events (#13200)

* Fixed an issue where some of the wallet state_changed events were not notified to subscribers

* Fixed a lint error

* Fix data Server CLI. (#13222)

* 🍒1 (#13231)

* add delayed pool config update feature (#12796)

* add delayed pool config update feature

* we dont talk about this commit

* We did not await the async function (#13154)

Co-authored-by: Jack Nelson <jack@jacknelson.xyz>
Co-authored-by: Mariano Sorgente <3069354+mariano54@users.noreply.github.com>

* update _peak_height after committing the block to the DB (#13157) (#13233)

Co-authored-by: Arvid Norberg <arvid@libtorrent.org>

* fill in data layer certs at startup (#13225)

* fill in data layer certs at startup

* tidy

* don't overwrite

* add overwrite parameter to write_ssl_cert_and_key()

* Change override to overwrite atari add_missing_files. (#13224)

* clarify comment in `DataLayerServer.stop()` (#13216)

* data layer - remove old .gitignore (#13198)

* correct port reference in data layer test (#13193)

* Update test_data_rpc.py

* catch up hashes with fix

* introduce an UncurriedPuzzle class (#13211)

* introduce an UncurriedPuzzle class, containin the mod and args from a puzzle that has already been uncurried. Pass this around to functions that want to access the uncurried puzzle, to avoid uncurrying the same puzzle multiple times.

* use UncurriedPuzzle for get_inner_puzzle()

* Calculate puzzle hashes directly, as an optimization. (#13034)

* Reflect the changes from main on atari related code.

* Implement generator_for_single_coin() in python instead of clvm and apply quex' optimization to only computing the puzzle hash if the parent and amount matches (#13237)

* Reflect the changes from main on atari related code.

* Apparently yaml.safe_dump() can't handle uint16 port (in the wallet_peer port section of data_layer's config). Propagate the port type and make this one explicitly int.

* Xch spam (#12878)

* Add mariano54s spam filter changes (original PR is #11894)

* Add new spam_filter tests

* remove get_all_coins method and query db directly in test

* Add test to include case where everything gets filtered, even the first coin

* If filter threshold is <= 1, no need to run the filter

* add test of chia.wallet.wallet_coin_store.get_all_unspent_coins

* Add test case for receiving large coin after sending dust coin

* Add NFT test to spam wallet

* resolve merge conflicts with main

* remove unneeded coin store test

* Add black formatting

* split long comments into multiple lines

* Run black, second try

* Manually add black changes

* Black

Co-authored-by: Mariano Sorgente <sorgente711@gmail.com>

* Pinning GUI to 8a52d3c - HEAD of release/1.6.0 (#13246)

* correct subscription ip to ipv4 localhost (#13243)

* data layer - wait for creation tx (#13240)

* DL Fix add_mirror amount -h. (#13221)

* unfix the test bug, we will get it later with the associated data changes

* coin interest optimization for trade records (#13044)

* coin interest optimization for trade records

* don't duplicate coin

* added a test

* migration test

* trade record replace test

* mypy fixes

* refactor to use coin_id

* return a set

* Get minter DID for offer (#13156)

* Get minter DID for offer

* Fetch minter did for all NFTs

* Resolve comments

* Fix issues

* Fix pre-commit

* Fix unit test

* update gui to checkpoint

* add wallet_id and did to state_changed (#13260)

* treehash optimization for DID wallet (#13253)

* only calculate treehash of constants once

* missed one treehash calc

* add benchmark for get_puzzle_and_solution_for_coin (#13301)

* fix: build_scripts/npm_linux/package.json & build_scripts/npm_linux/package-lock.json to reduce vulnerabilities (#13286)

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JS-GOT-2932019

* now when optimizations have landed, tighten the expected timing for the offer parsing benchmark (#13236)

* minor fixes for Serialized program (#13229)

* use bytearray to build a binary blob, convert it to bytes once it's done

* load the bootstrap generator rom in binary form directly, avoid conversions

* remove an insane amount of imports & add support for edge case (#13195)

add support for edge case

* fix flakey test (#12925)

* fix flakey test

* unused import

* testing flakeyness

* fix and test 200 times

* use mock to avoid running _sync

* lint

* remove unused import

* cmds: Key label support for CLI - `chia keys label` (#12917)

* Display minter DID in CLI (#13311)

* Get minter DID for offer

* Fetch minter did for all NFTs

* Resolve comments

* Fix issues

* Fix pre-commit

* Fix unit test

* Display minter did

* optimize get_block_store (#13263)

* optimize the block_store for the get_puzzle_and_solution_for_coin to not parse the full block. We just need the block generator and generator regs.

* optimize analyze-block

* address review comments

* astroid!=2.12.6, !=2.12.7 (#13338)

https://github.com/PyCQA/pylint/issues/7425

* Replace localhost with 127.0.0.1 for tests (#13324)

* Replace localhost with 127.0.0.1 in block tools config

* also make self_hostname fixture be 127.0.0.1

* Move config rewrite inside `automated_testing` conditional

* data layer - wait for creation tx (#13240) (#13335)

(cherry picked from commit 8b1a27b5ebb5fc9e584de3588199253f1da55df4)

* fix wallet rpc flakes  (#13336)

* fix flakes and lower timeouts

* lint

* Fix complex NFT offer flakiness (#13337)

* Load NFT off-chain metadata on backend (#13018)

* Fix NFT wallet naming

* Resolve comments

* Resolve comment

* Add tests

* Load off-chain metadata & nft_get_nfts pagination

* Fix pre-commit

* Resolve comments

* Resolve comments

* Fix precommit

* Resolve comments

* Resolve comment

* Fix pre-commit

* Fix unit test

* Add tests for pagination

* Resolve comments

* Fix pre-commit

* Resolve comments

* Ms.mempool simplify (#13314)

* Remove mempool.additions

* Don't re run the program, and remove program from mempool item

* Removals only stores item ids, and stores a list

* Move pending cache down to prevent cache dos

* Separate validation from adding to pool, and remove mypy exceptions

* Fix bug with replacing

* Add to mypy

* Revert cbgui

* precommit fail

* Properly update the seen dict

* lint error

* Fix mempool bug

* Update after merge with main

* Address comments

* Fix bug in remove_plot_directory (#13332)

Revert commit c400d816afa7a8a54cf872deb787aaa3a144363b

* Avoid creating a list and enable short circuit behavior in `bundle_suitable_for_compression()` (#13331)

This avoids the creation of the list entirely.  In so doing it also allows `all()` to return `False` on the first uncompressable spend it finds instead of analyzing all of them first.

* Remove "Total iterations since start" from chia show -s (#13196)

* Bumping codeQL actions to v2 (#13344)

* use get_puzzle_and_solution_for_coin() from chia_rs (#13315)

* The poor CI machine cannot go fast enough (#13347)

* The poor CI machine cannot go fast enough

* Also for data layer

* More increases

* Update build-linux-installer-deb.yml

* Fixing bladebit plotter conditional

Co-authored-by: Amine Khaldi <amine.khaldi@reactos.org>
Co-authored-by: Kyle Altendorf <sda@fstab.net>
Co-authored-by: Florin Chirica <fchirica96@gmail.com>
Co-authored-by: Jeff <jeff@chia.net>
Co-authored-by: dustinface <35775977+xdustinface@users.noreply.github.com>
Co-authored-by: Matt Hauff <quexington@gmail.com>
Co-authored-by: Arvid Norberg <arvid@libtorrent.org>
Co-authored-by: Mariano Sorgente <3069354+mariano54@users.noreply.github.com>
Co-authored-by: Justin England <justin@chia.net>
Co-authored-by: Izumi Hoshino <admin@chiamine.jp>
Co-authored-by: hugepants <hugepants@users.noreply.github.com>
Co-authored-by: Chris Marslender <chrismarslender@gmail.com>
Co-authored-by: Jack Nelson <jack@jacknelson.xyz>
Co-authored-by: Richard Kiss <him@richardkiss.com>
Co-authored-by: Dan Perry <69756004+danieljperry@users.noreply.github.com>
Co-authored-by: Mariano Sorgente <sorgente711@gmail.com>
Co-authored-by: Sebastjan Trepca <trepca@gmail.com>
Co-authored-by: Kronus91 <t.yu@chia.net>
Co-authored-by: Snyk bot <github+bot@snyk.io>
Co-authored-by: Almog De Paz <almogdepaz@gmail.com>
Co-authored-by: joshpainter <josh@joshpainter.com>
Co-authored-by: Gene Hoffman <30377676+hoffmang9@users.noreply.github.com>

wallentx committed:
Adding installer uploads on release

wallentx committed:
Fixing artifact upload path

wallentx committed:
Fixing path to artifact upload

wallentx committed:
Adding torrent uploads

wallentx committed:
blob file matching for artifact upload
